src/flavors/list.rs

```diff
@@ -195,20 +218,20 @@ impl<T> Channel<T> {
                     ).is_ok()
                 {
                     // If this was the last slot in the block, install a new block.
-                    if offset + 1 == BLOCK_CAP {
+                    if offset + 1 == tail_slot_count {
                         install_next_block();
                     }
 
                     unsafe {
-                        let slot = tail.slots.get_unchecked(offset).get();
-                        (*slot).msg.get().write(ManuallyDrop::new(msg));
+                        let slot = tail.slots.get_unchecked(offset);
+                        *slot.msg.get() = Some(msg);
                         (*slot).ready.store(true, Ordering::Release);
                     }
                     break;
                 }
 
                 backoff.spin();
-            } else if offset == BLOCK_CAP {
+            } else {
```

Comment:
Let's keep the condition: `} else if offset == tail_slot_count {`

Reply:
I understand being explicit here and I can add it back in, but don't we know based off the way that `new_slot_count` is calculated that if it's not less than, then it must be equal?

Reply:
No, it can also be greater. :)

Note that we first load the block and then load the index. Between these two moments, there's nothing stopping the index from progressing beyond `tail_start_index + tail_slot_count`.

Reply:
Ok I see the importance of the load `Ordering` there